xarray/core/dataset.py

```diff
@@ -1204,9 +1277,22 @@ def sel(self, method=None, tolerance=None, drop=False, **indexers):
         Dataset.isel_points
         DataArray.sel
         """
+        from .dataarray import DataArray
+
+        new_coords = {k: v._coords for k, v in indexers.items()
+                      if isinstance(v, DataArray)}
+
+        indexers = {k: v.variable if isinstance(v, DataArray) else v
```

Comment:
Use another name than `indexers` here for the assigned variable. That way you can still look  things up on the original `indexers` arguments rather than saving `new_coords` separately.

Reply:
Done.